src/libs/actions/Task.js

```diff
@@ -0,0 +1,158 @@
+import Onyx from 'react-native-onyx';
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import * as ReportUtils from '../ReportUtils';
+import * as Report from './Report';
+import Navigation from '../Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+
+let currentUserEmail;
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (val) => {
+        // When signed out, val is undefined
+        if (!val) {
+            return;
+        }
+
+        currentUserEmail = val.email;
+    },
+});
+
+/**
+ * Clears out the task info from the store
+ */
+function clearOutTaskInfo() {
+    Onyx.set(ONYXKEYS.TASK, null);
+}
+
+/**
+ * Assign a task to a user
+ * Function name is createTask for consistency with the rest of the actions
+ * and also because we can create a task without assigning it to anyone
+ * @param {String} parentReportID
+ * @param {String} name
+ * @param {String} description
+ * @param {String} assignee
+ *
+ */
+
+function createTaskAndNavigate(parentReportID, name, description, assignee) {
+    // See if there's already a chat report with the assignee
+    const assigneeChatReportID = ReportUtils.getChatByParticipants([assignee]).reportID;
+
+    // Create the CreatedReportAction on the parent chat report
+    const optimisticCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(parentReportID);
```

Comment:
Should not need this actually, tried to create a flow chart in the slack but because we're creating any optimistic reports between the DM creator and the assignee separately we shouldn't need to add this createdAction to anything except the task

Reply:
Okay, I think I've been thinking about the createdAction all wrong. In my mind it was for the parent, but now I understand it's for the optimistically created task report.